### ISSUE TITLE
Add support for tar with bzip compression

### ISSUE THREAD

**Issue by zulus** [eclipse-platform/eclipse.platform.ui#1140]
I realized that PDT contain copy/paste version of TarLeveledStructureProvider and TarFile only because platform.ui version doesn't suport CBZip2InputStream  (tar.bz):  https://github.com/eclipse-pdt/pdt/blob/2c5def3c0195bf5b49008ebe225e5f9d6607cbf4/plugins/org.eclipse.php.core/src/org/eclipse/php/internal/core/tar/TarFile.java#45

I would like to move this part into platform.ui

**Comment by BeckerWdf** [comment]
@zulus: Sure why not? Pls. provide a PR for this feature. See: https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md

**Comment by zulus** [comment]
Currently PDT have built-in BZIP2 support. But maybe will be better to migrate to Apache Commons-Compress and forgot about maintaining?

**Comment by mickaelistria** [comment]
> But maybe will be better to migrate to Apache Commons-Compress and forgot about maintaining?

Yes, the less things we maintain, the more we reuse, and the better job we do  IMO.
